monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -2174,12 +2176,17 @@ object Task extends TaskInstancesLevel1 {
   def shift(ec: ExecutionContext): Task[Unit] =
     Async[Unit] { (context, cb) =>
       val ec2 = if (ec eq null) context.scheduler else ec
-      ec2.execute(new Runnable {
-        def run(): Unit = {
-          context.frameRef.reset()
-          cb.onSuccess(())
-        }
-      })
+      try {
+        ec2.execute(new Runnable {
+          def run(): Unit = {
+            context.frameRef.reset()
+            cb.onSuccess(())
+          }
+        })
+      } catch {
+        case ex: RejectedExecutionException =>
+          cb.onError(ex)
```

Comment:
Works, but use `cb.asyncOnError(ex)` as it's safer — I make it a habit of doing that in `Async` states in order to ensure stack safety, as you never know how it ends up being used. It's not needed for `onSuccess` here because you have `ec2.execute` first.